### PR TITLE
Remove local env variables

### PR DESCRIPTION
**Pull request by benjamin-shen**
## Summary
Added default values and refactored `process.env` such that local env variables are unnecessary for single-tenant hosting (i.e. for open-source contributors that want to make changes to client). `IS_MULTI_TENANT=true` and `ADMIN_TOKEN` are still required for multi-tenant hosting.

After this PR is merged, devs will be able to simplify their local `.env` files and there will be reduced friction to start contributing to `client/` :)

## Test Plan
1. Go into the `client/` folder.
2. Comment out your entire `.env` file.
3. Run `yarn preconfigure`, then `yarn dev`. Navigate to `http://localhost:3000` and sanity check.
4. Add back `IS_MULTI_TENANT=true` and `ADMIN_TOKEN` (but not `API_ENDPOINT`) to your `.env` file.
5. Start server locally. Navigate to `http://mintlify.localhost:3000` and sanity check. Validate `sitemap.xml` and `robots.txt`.

## TODO
- [ ] Reorganize/remove `lib/` folder and remove exception from `.gitignore`
- [ ] Clean up Infisical secrets in Mintlify Client
- [ ] Clean up Vercel env variables
- [ ] Add instructions (internal docs) for multi-tenant local hosting

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -21,9 +21,9 @@ yarn
 
 ### Local Development
 
-- Log in to Infiscal for the .env variables. The project is `Mintlify Client`. Save those env vars to `client/.env`.
+- Run `yarn preconfigure` or `yarn preconfigure <path/to/docs>`
 - Run `yarn dev`
-- Navigate to `http://mintlify.localhost:3000/`
+- Navigate to `http://localhost:3000/`
```
Comment (hahnbeelee) [inline client/README.md:26]: This makes 100% sense for open source contributors but the old version helps a ton for internal devs. Maybe we should just write internal docs at least specifically for mint. Too many different stakeholders are looking at this repo.

File: client/src/constants.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const API_ENDPOINT = process.env.API_ENDPOINT || 'http://localhost:5000';
```
Comment (hahnbeelee) [inline client/src/constants.ts:1]: this file is chef's kiss

File: client/src/data-fetching/getHiddenPages.ts [hunk 3]
```diff
@@ -1,10 +1,4 @@
...
-      headers: { Authorization: `Bearer ${process.env.ADMIN_TOKEN}` },
-    }
-  );
-  const hiddenPages = await hiddenPagesResponse.json();
-  return hiddenPages;
-};
+import { fetchWithAuth } from './fetch';
+
+export const getHiddenPages = async (subdomain: string): Promise<string[]> =>
+  (await fetchWithAuth(subdomain, '/hidden-pages')).json();
```
Comment (hahnbeelee) [inline client/src/data-fetching/getHiddenPages.ts:4]: dis is so clean

**Comment by hahnbeelee** [comment]
What about INTERNAL_ANALYTICS_WRITE_KEY?

**Review by hahnbeelee — APPROVED** [review]
- [x] Successful test plan

We gotta improve internal documentation

File: client/README.md (hunk 1 above)
Comment (hahnbeelee) [reply client/README.md:26]: Oh I understand - this is your last todo list item

Comment (benjamin-shen) [reply client/README.md:26]: The old instructions also were incomplete on the backend side. We should keep all the public documentation relevant to open-source

**Comment by benjamin-shen** [comment]
> What about `INTERNAL_ANALYTICS_WRITE_KEY`?

I believe that's only used in production, right? No need for local development

File: client/README.md (hunk 1 above)
Comment (hahnbeelee) [reply client/README.md:26]: sounds good

**Comment by hahnbeelee** [comment]
> I believe that's only used in production, right? No need for local development

Yeah that's true

**Comment by hahnbeelee** [comment]
> I believe that's only used in production, right? No need for local development

Wait actually I don't really view this as "removing" local env variables, I kind of just view it as a refactor to consolidate all the places we do use an environment variable to make the code cleaner. All the environment variables in `constants.ts` are used for enterprise mintlify features (not self-hostable)

**Review by hahnbeelee — APPROVED** [review]
Did some white box testing. LGTM

**Pull request merged**